src/DqtApi/DataStore/Crm/DataverseAdapter.UpdateTeacher.cs

```diff
@@ -0,0 +1,563 @@
+            }
+
+            public (dfeta_qualification Result, UpdateTeacherFailedReasons? FailedReason) SelectQualificationRecord(
+            IEnumerable<dfeta_qualification> qualificationRecords)
+            {
+                // Find an Qualification for provider.
+                var qualificationForProvider = qualificationRecords
+                    .ToArray();
+
+                if (qualificationForProvider.Length == 1)
+                {
+                    return (qualificationForProvider[0], null);
+                }
+                else if (qualificationForProvider.Length > 1)
+                {
+                    return (null, UpdateTeacherFailedReasons.MultipleQualificationRecords);
+                }
+                else
+                {
+                    return (null, null);
+                }
+            }
+
+            public CrmTask CreateReviewTaskEntity(UpdateTeacherFindResult teacher)
+            {
+                var description = GetDescription();
+
+                return new CrmTask()
+                {
+                    RegardingObjectId = new EntityReference(Contact.EntityLogicalName, TeacherId),
+                    Category = "DMSImportTrn",
+                    Subject = "Notification for QTS Unit Team",
+                    Description = description,
+                    ScheduledEnd = _dataverseAdapter._clock.UtcNow
+                };
+
+                string GetDescription()
+                {
+                    var sb = new StringBuilder();
+                    sb.Append($"Active sanction found: TRN {TeacherId} and date of birth {teacher.DateOfBirth}");
+                    return sb.ToString();
+                }
+            }
+
+            public CrmTask CreateQualificationReviewTaskEntity(UpdateTeacherFindResult teacher)
+            {
+                var description = GetDescription();
+
+                return new CrmTask()
+                {
+                    RegardingObjectId = new EntityReference(Contact.EntityLogicalName, TeacherId),
+                    Category = "DMSImportTrn",
+                    Subject = "More than one qualification record found",
+                    Description = description,
+                    ScheduledEnd = _dataverseAdapter._clock.UtcNow
+                };
+
+                string GetDescription()
+                {
+                    var sb = new StringBuilder();
+                    sb.Append($"More than one qualification record found");
+                    return sb.ToString();
+                }
+            }
+
+            public CrmTask CreateIttReviewTask(UpdateTeacherFindResult teacher, string message)
+            {
+                var description = GetDescription();
+
+                return new CrmTask()
+                {
+                    RegardingObjectId = new EntityReference(Contact.EntityLogicalName, TeacherId),
+                    Category = "DMSImportTrn",
+                    Subject = message,
```

Comment:
Should Subject and Description both be the same here?

Reply:
I've set it to that of Create teacher for now, Tusa is looking into this currently.